File: client/src/components/Reports/controlPanels/Configuration/AggregationType.js
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import {Select} from 'components';
import {t} from 'translation';
import {isDurationReport} from 'services';

export default function AggregationType({report, onChange}) {
  const {data} = report;

  const isVariableReport = data?.view?.entity === 'variable';

  if (isDurationReport(report) || isVariableReport) {
    return (
      <fieldset className="AggregationType">
        <legend>{t('report.config.aggregation.legend')}</legend>
        <Select
          value={data.configuration.aggregationType}
          onChange={(value) => onChange({aggregationType: {$set: value}}, true)}
        >
          {isVariableReport && (
            <Select.Option value="sum">{t('report.config.aggregation.sum')}</Select.Option>
          )}
          <Select.Option value="min">{t('report.config.aggregation.minimum')}</Select.Option>
          <Select.Option value="avg">{t('report.config.aggregation.average')}</Select.Option>
          {!data.configuration.processPart && (
            <Select.Option value="median">{t('report.config.aggregation.median')}</Select.Option>
          )}
          <Select.Option value="max">{t('report.config.aggregation.maximum')}</Select.Option>
        </Select>
      </fieldset>
    );
  }
  return null;
}

File: util/optimize-commons/src/main/java/org/camunda/optimize/dto/optimize/query/report/single/configuration/AggregationType.java
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under one or more contributor license agreements.
 * Licensed under a proprietary license. See the License.txt file for more information.
 * You may not use this file except in compliance with the proprietary license.
 */
package org.camunda.optimize.dto.optimize.query.report.single.configuration;

import com.fasterxml.jackson.annotation.JsonValue;

import static org.camunda.optimize.dto.optimize.ReportConstants.AVERAGE_AGGREGATION_TYPE;
import static org.camunda.optimize.dto.optimize.ReportConstants.MAX_AGGREGATION_TYPE;
import static org.camunda.optimize.dto.optimize.ReportConstants.MIN_AGGREGATION_TYPE;
import static org.camunda.optimize.dto.optimize.ReportConstants.PERCENTILE_AGGREGATION_TYPE;
import static org.camunda.optimize.dto.optimize.ReportConstants.SUM_AGGREGATION_TYPE;

public enum AggregationType {

  AVERAGE(AVERAGE_AGGREGATION_TYPE),
  MIN(MIN_AGGREGATION_TYPE),
  MAX(MAX_AGGREGATION_TYPE),
  SUM(SUM_AGGREGATION_TYPE),
  PERCENTILE(PERCENTILE_AGGREGATION_TYPE);

  private final String id;

  AggregationType(final String id) {
    this.id = id;
  }

  @JsonValue
  public String getId() {
    return id;
  }

  @Override
  public String toString() {
    return getId();
  }

}
